vmm/src/lib.rs

```diff
@@ -1133,14 +1133,22 @@ impl Vmm {
                 info!("Migration complete");
                 Ok(())
             } {
-                // Stop logging dirty pages and keep the source VM paused unpon successful migration
+                // Stop logging dirty pages and shutdown the source VM paused unpon successful migration
                 Ok(()) => {
                     // Let every Migratable object know about the migration being complete
                     vm.migration_complete()?;
 
                     // Stop logging dirty pages
                     vm.stop_dirty_log()?;
 
+                    // Shutdown the VM after the migration succeeded
```

Comment:
```
vmm: Shutdown VM after migration succeeded

In case the migration succeeds, the destination VM will be correctly
running, with potential vhost-user backends attached to it. We can't let
the source VM trying to reconnect to the same backends, which is why
it's safer to shutdown the source VM.
```
The idea is that when performing local migration, the same socket will be used by the destination VM, which means we can't have two VMs trying to connect to the same backend concurrently. Therefore, once the migration is complete, we can't let the source VM try to reconnect to the backend, and can't let it run without a functional vhost-user backend. Since there's no solution to this problem, we must shutdown the VM, otherwise we would leave our users with a partially functional VM.

I also think we gotta wait and see when our users will really use live migration. We'll listen to their inputs and make some changes if we/they are not happy with the current behavior.

Reply:
I think this is the correct approach.

Reply:
Okay. I see. I assumed the actions of closing the vu threads and deleting the "vu" handle were enough to stop the source VM reconnecting the backend, which turned to be a wrong understanding. Thanks for the explanation.

Reply:
Okay. I see. I assumed the actions of closing the vu threads and deleting the "vu" handle were enough to stop the source VM reconnecting the backend, which turned to be a wrong understanding. Thanks for the explanation.